end_to_end/include/action.hpp

```diff
@@ -191,12 +200,17 @@ class TestActionConnect : public TestAction {
     return std::shared_ptr<TestActionConnect>(new TestActionConnect());
   }
 
-  void execute_unchecked(const std::string& case_name) const override {
+  auto execute_unchecked(const std::string& case_name) const
+      -> AstarteDeviceSdk::astarte_tl::expected<void, AstarteError> override {
     spdlog::info("[{}] Connecting...", case_name);
-    device_->connect();
+    auto res = device_->connect();
+    if (!res) {
+      return res;
+    }
     do {
       std::this_thread::sleep_for(std::chrono::seconds(1));
     } while (!device_->is_connected());
+    return {};
```

Comment:
I was thinking of a way to avoid using the `if(!ret){...}` block. What do think of this:
```suggestion
return device_->connect().transform([this]() {
      do {
        std::this_thread::sleep_for(std::chrono::seconds(1));
      } while (!device_->is_connected());
    });
```

We can exploit transform so that when it fails it immediatelly return the error (thus replacing the if block), whereas if it doesn't fail it waits until the device is connected